### PR TITLE
chore(ci): tweaks coverage.

### PR DESCRIPTION
**Pull request by jcchavezs**
This PR upgrades the coverage tool and decreases the number of builds.

Thanks to @peter279k

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -25,29 +24,27 @@ jobs:
...
           restore-keys: ${{ runner.os }}-composer-
       - name: Install Composer dependencies
         run: |
           composer install --no-progress --prefer-dist --optimize-autoloader
       - name: Download coveralls.phar
-        if: matrix.operating-system != 'windows-latest'
+        if: matrix.operating-system == 'ubuntu-latest'
         run: |
-          wget -c -nc --retry-connrefused --tries=0 https://github.com/satooshi/php-coveralls/releases/download/v1.0.1/coveralls.phar
+          wget -c -nc --retry-connrefused --tries=0 https://github.com/satooshi/php-coveralls/releases/download/v2.4.2/coveralls.phar
```
Comment (peter279k) [inline .github/workflows/ci.yml:35]: It should be `https://github.com/php-coveralls/php-coveralls/releases/download/v2.4.2/php-coveralls.phar`.

Comment (jcchavezs) [inline .github/workflows/ci.yml:35]:
```suggestion
          wget -c -nc --retry-connrefused --tries=0 https://github.com/satooshi/php-coveralls/releases/download/v2.4.2/php-coveralls.phar
```

Comment (jcchavezs) [reply .github/workflows/ci.yml]: Thanks!

File: tests/Unit/Reporters/JsonV2SerializerTest.php [hunk 2]
```diff
@@ -85,11 +85,9 @@ public function testJSONTagsAreSerializedCorrectly()
         $expectedSerialization = '[{'
             . '"id":"186f11b67460db4e","traceId":"186f11b67460db4e","timestamp":1594044779509687,"name":"test",'
             . '"duration":1000,"localEndpoint":{"serviceName":"service1"},'
-            . '"tags":{"test_key_1":"{\"name\":\"Kurt\"}","test_key_2":"foo\nbar"}'
+            . '"tags":{"test_key_1":"{\"name\":\"Kurt\"}","test_key_2":"foo' . \PHP_EOL . 'bar"}'
```
Comment (jcchavezs) [inline tests/Unit/Reporters/JsonV2SerializerTest.php:88]: I am curious about this @adriancole: when a tag contains linebreaks and when we escape them and turn linebreaks into \n and for Linux or MacOS, what do we do for windows, do we turn into \n too or \r\n?

**Comment by peter279k** [comment]
After looking at the latest GitHub Action log, it seems that the `coveralls` coverage uploading work has been failed.

The reason that `php-coveralls` needs the `COVERALLS_REPO_TOKEN` variable on GitHub Action.

More detailed references are as follows:

- https://github.com/php-coveralls/php-coveralls/issues/273
- https://github.com/php-coveralls/php-coveralls/issues/301
- https://github.com/php-coveralls/php-coveralls/issues/300

@jcchavezs, I cannot help you to work on that because it's up to you to setup a secret coveralls token.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -25,29 +24,27 @@ jobs:
...
-      - name: Run Tests
+      - name: Run tests
         run: composer test -- --coverage-clover=build/logs/clover.xml
       - name: Upload coverage report to codecov service
-        if: matrix.operating-system != 'windows-latest'
+        if: matrix.operating-system == 'ubuntu-latest'
         run: |
-          php coveralls.phar -v
+          php php-coveralls.phar -v
           bash <(curl -s https://codecov.io/bash)
```
Comment (jcchavezs) [inline .github/workflows/ci.yml:50]:
```suggestion
          bash <(curl -s https://codecov.io/bash)
        env:
          COVERALLS_RUN_LOCALLY: 1
          COVERALLS_REPO_TOKEN: ${{ secrets.COVERALLS_REPO_TOKEN }}

```

**Pull request merged**